src/librustc/ty/sty.rs

```diff
@@ -141,6 +141,10 @@ pub enum TypeVariants<'tcx> {
     /// `|a| yield a`.
     TyGenerator(DefId, ClosureSubsts<'tcx>, GeneratorInterior<'tcx>),
 
+    /// A type representin the types stored inside a generator.
+    /// This should only appear in GeneratorInteriors.
+    TyGeneratorWitness(Binder<&'tcx Slice<Ty<'tcx>>>),
```

Comment:
I'm a bit confused about the role of this binder. I remember that, in the current generator code, there is a binder around the signature that doesn't much of a role at all -- I had thought about removing it -- but I take it that this binder is somewhat different.

Reply:
I guess I mean: what regions are being bound in this binder?

Reply:
OK, I see the answer.